### PR TITLE
 Signer functionality 

### PR DESCRIPTION
**Pull request by TomasArrachea**
This PR adds a signer crate to handle Signer instantiation.

### CODE REVIEW

File: crates/signer/src/signer.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    #[error("invalid private key")]
+    InvalidPrivateKey,
+    #[error("invalid keystore password")]
+    InvalidPassword,
+}
+
+impl Config {
+    /// Creates a signer from given config.
+    pub fn signer_from_config(c: Config) -> Result<PrivateKeySigner, SignerError> {
+        // TODO: check chain id to select signer
```
Comment (TomasArrachea) [inline crates/signer/src/signer.rs:29]: Check if an equivalent of [LatestSignerForChainID](https://github.com/ethereum/go-ethereum/blob/master/core/types/transaction_signing.go#L91) exists for rust

File: crates/signer/src/signer.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
+use alloy_signer_local::PrivateKeySigner;
+use eth_keystore::decrypt_key;
+use std::path::Path;
+use thiserror::Error;
+
+#[derive(Debug)]
+/// Represents the input params to create a signer
```
Comment (dralves) [inline crates/signer/src/signer.rs:7]: nit can you put the docs above the derive? here and else where

File: crates/signer/src/signer.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    use alloy_primitives::{address, bytes, U256};
+    use alloy_signer::Signature;
+    use alloy_signer_local::PrivateKeySigner;
+    use hex_literal::hex;
+    use std::str::FromStr;
+
+    #[test]
+    fn sign_transaction_with_private_key() {
+        let private_key =
+            "dcf2cbdd171a21c480aa7f53d77f31bb102282b3ff099c78e3118b37348c72f7".to_owned();
```
Comment (dralves) [inline crates/signer/src/signer.rs:62]: pull a constant? (same below)

**Review by dralves — APPROVED** [review]
approved barring the minor nits

File: crates/signer/src/signer.rs (hunk 2 above)
Comment (ricomateo) [reply crates/signer/src/signer.rs:6]: Done!

File: crates/signer/src/signer.rs (hunk 3 above)
Comment (ricomateo) [reply crates/signer/src/signer.rs]: Done!

**Review by dralves — APPROVED** [review]

**Pull request merged**